### ISSUE TITLE
F1Confidence と PRCurve plotter の precision 処理を共通化

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#601]
## Summary

`F1ConfidencePlotter` と `PRCurvePlotter` で precision テンソルの clean 処理 (`valid_mask = precision >= 0` → NaN 置換 → クラスループ) が重複している. `_clean_precision_tensor()` ヘルパーに抽出する.

## Background

両 plotter は同じ COCO 評価結果テンソルから precision を取り出してプロットする. 無効値マスクと NaN 置換のロジックがほぼ字面同一で, 両者を保守する際に同じ修正を 2 度行う必要がある.

## Current Implementation

- [pochidetection/visualization/plotters/f1_confidence_plotter.py](pochidetection/visualization/plotters/f1_confidence_plotter.py) の precision 処理 (lines 79-95, 172-234 周辺)
- [pochidetection/visualization/plotters/pr_curve_plotter.py](pochidetection/visualization/plotters/pr_curve_plotter.py) の precision 処理 (lines 80-122, 124-180 周辺)

両者とも以下のパターン:

```python
valid_mask = precision_slice >= 0
precision_clean = torch.where(valid_mask, precision_slice, torch.full_like(precision_slice, float("nan")))
# クラスループで precision_clean[valid] フィルタリング
```

## Proposed Changes

`pochidetection/visualization/plotters/_precision_utils.py` (新規) または既存共通モジュールに以下を追加:

```python
def clean_precision_tensor(precision: torch.Tensor) -> torch.Tensor:
    """無効値 (precision < 0) を NaN に置換した clean テンソルを返す."""
    ...
```

両 plotter からこのヘルパーを呼ぶ.

## Acceptance Criteria

- precision clean ロジックが共通ヘルパーに集約される
- F1ConfidencePlotter と PRCurvePlotter からの直接呼び出しに置き換わる
- 既存テストが pass する

## Labels

`refactor`

## Branch

`refactor/`
